chunk/chunk.go

```diff
@@ -307,6 +313,9 @@ func (s *ValidatorStore) Put(ctx context.Context, mode ModePut, chs ...Chunk) (e
 func (s *ValidatorStore) validate(ch Chunk) bool {
 	for _, v := range s.validators {
 		if v.Validate(ch) {
+			if s.pssCallback != nil {
+				go s.pssCallback(ch) // TODO: needs error handling
```

Comment:
This only needs to be called on 
* content address chunks
* that are close to the node address

On second thought it could be in the validator.

food for thought: if we receive the chunk twice, we call the callback twice. this shouldn't happen right?
if chunk is already in  localstore, then no callback

Reply:
> This only needs to be called on
> 
>* content address chunks
>* that are close to the node address

i was not aware that `ValidatorStore.validate` could be called in these cases (is this also true for `ValidatorStore.Put`?)

i'm wondering if it's too much of a hassle to check this here. is there a place in the code where these conditions are already met?

> On second thought it could be in the validator.

do you mean injecting the callback into a `Validator` object instead of the `ValidatorStore`?
 
> food for thought: if we receive the chunk twice, we call the callback twice. this shouldn't happen right?
> if chunk is already in localstore, then no callback

same here as in the first point: i'll try to find a place in the code where this is already true, so that it does not have to be verified.

Reply:
created issue #2163 for this